static/conformance-view.js

```diff
@@ -0,0 +1,314 @@
+
+    require('selectize');
+
+    var Compiler = require('./compiler');
+
+    var compilers = options.compilers;
+    var compilersById = {};
+    _.forEach(compilers, function (compiler) {
+        compilersById[compiler.id] = compiler;
+        if (compiler.alias) compilersById[compiler.alias] = compiler;
+    });
+
+    function Conformance(hub, container, state) {
+        this.container = container;
+        this.eventHub = hub.createEventHub();
+        this.domRoot = container.getElement();
+        this.domRoot.html($('#conformance').html());
+        this.selectorList = this.domRoot.find('.compiler-list');
+        this.addCompilerButton = this.domRoot.find('.add-compiler');
+        this.compileButton = this.domRoot.find('.compile');
+        this.editorId = state.editorid;
+        this.source = state.source || "";
+        this.currentExpandedSource = "";
+        this.nextSelectorId = 0;
+
+        this.maxCompilations = 6;
+
+        this.status = {
+            allowCompile: false,
+            allowAdd: true
+        };
+
+        this.container.on('destroy', function () {
+            this.eventHub.emit("conformanceViewClose", this.editorId);
+            this.eventHub.unsubscribe();
+        }, this);
+
+        this.container.on('open', function () {
+            this.eventHub.emit("conformanceViewOpen", this.editorId);
+        }, this);
+
+        this.eventHub.on('editorChange', this.onEditorChange, this);
+        this.eventHub.on('editorClose', this.onEditorClose, this);
+
+        this.addCompilerButton.on('click', _.bind(function() {
+            this.addCompilerSelector();
+            this.saveState();
+        }, this));
+
+        this.compileButton.on('click', _.bind(function() {
+            this.selectorList.find('.status').css("visibility", "hidden");
+            this.compileAll();
+        }, this));
+
+        if (state.compilers) {
+            _.each(state.compilers, _.bind(function (config) {
+                config.cv = this.nextSelectorId;
+                this.nextSelectorId++;
+                this.addCompilerSelector(config);
+            }, this));
+        }
+
+        this.handleToolbarUI();
+    }
+
+    Conformance.prototype.setTitle = function (compilerCount) {
+        this.container.setTitle("Conformance viewer (Editor #" + this.editorId + ") " + (
+                compilerCount !== 0 ? (compilerCount + "/" + this.maxCompilations) : ""
+            ));
+    };
+
+    Conformance.prototype.addCompilerSelector = function(config) {
+        if (!config) {
+            config = {
+                // Have we been compiled? If so, what is the result?
+                status: {
+                    // 0: Not compiled; 1: Ok; 2: Warnings; 3: Error
+                    code: 0,
+                    // If compiled, what the compiled output to stdout/err
+                    text: ""
+                },
+                // Code we have
+                cv: this.nextSelectorId,
```

Comment:
Kinda. When loading a shared state, cv are reset and `nextSelectorId` set to the max + 1